File: GO/twosum.go
package main

import "fmt"

type solution struct {
	currentIndex int
	innerIndex   int
}

func twoSum(this *solution, nums [4]int, target int) [2]int {
	this.currentIndex = 0
	this.innerIndex = 0
	final := [2]int{-1, -1}

	for _, number := range nums {
		for _, i := range nums {
			if this.currentIndex != this.innerIndex {
				if number+i == target {
					final[0] = this.currentIndex
					final[1] = this.innerIndex
					return final
				}
			}
			this.innerIndex++
		}
		this.currentIndex++
		this.innerIndex = 0
	}
	return final
}

func main() {
	s := solution{}

	nums := [4]int{2, 7, 11, 15}
	target := 9

	answer := twoSum(&s, nums, target)

	fmt.Println(answer)
}

File: CPP/twosum.cpp
#include <iostream>
#include <vector>

class Solution
{
private:
  int m_currentIndex;
  int m_innerIndex;

public:
  Solution() : m_currentIndex(0), m_innerIndex(0) {}
  std::vector<int> twoSum(const std::vector<int> &nums, const int target);
};

int main()
{
  Solution s;

  std::vector<int> input = {2, 7, 11, 15};
  int target = 9;

  auto answer = s.twoSum(input, target);

  std::cout << "Answer: " << answer[0] << ", " << answer[1] << std::endl;
}

std::vector<int> Solution::twoSum(const std::vector<int> &nums, const int target)
{
  this->m_currentIndex = 0;
  this->m_innerIndex = 0;
  std::vector<int> final = {-1, -1};

  for (int number : nums)
  {
    for (int i : nums)
    {
      if (this->m_currentIndex != this->m_innerIndex)
      {
        if (number + i == target)
        {
          final = {this->m_currentIndex, this->m_innerIndex};
          return final;
        }
      }
      this->m_innerIndex++;
    }
    this->m_currentIndex++;
    this->m_innerIndex = 0;
  }

  return final;
}